crates/ruff_linter/src/rules/pylint/rules/self_or_cls_assignment.rs

```diff
@@ -47,10 +53,14 @@ impl Violation for SelfOrClsAssignment {
         let SelfOrClsAssignment { method_type } = self;
 
         format!(
-            "Invalid assignment to `{}` argument in {method_type} method",
+            "Confusing assignment to `{}` argument in {method_type} method",
             method_type.arg_name(),
```

Comment:
Another option could be "Re-assigned `{}` argument in {method_type} method"

Reply:
Oh, that's better. Much more descriptive. I'll make another PR 😄